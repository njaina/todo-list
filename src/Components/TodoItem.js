const TodoItem = ({ todo, handleCheck, index }) => (
    <div className="response">
      <div className="element" key={todo}>
        {todo}
      </div>
      <div className="check">
        <input type="checkbox" onClick={() => handleCheck(index)} />
      </div>
    </div>
  );
  export default TodoItem;