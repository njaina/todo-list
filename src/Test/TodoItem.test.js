import React from 'react';
import renderer from 'react-test-renderer';
import TodoItem from '../Components/TodoItem';

describe('TodoItem component', () => {
  it('renders correctly', () => {
    const todo = 'This is a sample todo';
    const handleCheck = jest.fn();
    const index = 0;

    const tree = renderer
      .create(<TodoItem todo={todo} handleCheck={handleCheck} index={index} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
