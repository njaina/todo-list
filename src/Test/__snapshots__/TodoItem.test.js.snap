// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem component renders correctly 1`] = `
<div
  className="response"
>
  <div
    className="element"
  >
    This is a sample todo
  </div>
  <div
    className="check"
  >
    <input
      onClick={[Function]}
      type="checkbox"
    />
  </div>
</div>
`;
